test/jdk/sun/tools/jcmd/TestJcmdNoAgentLoad.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import static jdk.test.lib.Asserts.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8304438
+ * @summary JVMTI.agent_load should obey EnableDynamicAgentLoading
+ *
+ * @library /test/lib
+ *
+ * @run main/othervm -XX:-EnableDynamicAgentLoading TestJcmdNoAgentLoad
+ */
+public class TestJcmdNoAgentLoad {
+    // The Agent.jar does not exist. Error with the PTRN message
+    // is expected before an attempt to load the Agent.jar.
+    private static final String[] CMD = new String[] { "JVMTI.agent_load", "Agent.jar" };
+    private static final String PTRN = "Dynamic agent loading is not enabled";
+
+    public static void main(String[] args) throws Exception {
+        testNoAgentLoad(CMD);
+    }
+
+    private static void testNoAgentLoad(String... jcmdArgs) throws Exception {
+        OutputAnalyzer output = JcmdBase.jcmd(jcmdArgs);
+
+        assertEquals(output.getExitValue(), 0);
```

Comment:
I think tests normally just use `output.shouldHaveExitValue(0);`

Reply:
Thank you for the suggestion. I've derived this code from one of the existing jcmd tests.